NotificationApi/NotificationApi.DAL/Services/CreateNotificationService.cs

```diff
@@ -0,0 +1,41 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using NotificationApi.Common;
+using NotificationApi.DAL.Commands;
+using NotificationApi.DAL.Commands.Core;
+using NotificationApi.DAL.Queries;
+using NotificationApi.DAL.Queries.Core;
+using NotificationApi.Domain;
+using Notify.Interfaces;
+
+namespace NotificationApi.DAL.Services
+{
+    public class CreateNotificationService : ICreateNotificationService
+    {
+        private readonly IAsyncNotificationClient _asyncNotificationClient;
+        private readonly ICommandHandler _commandHandler;
+        private readonly IQueryHandler _queryHandler;
+
+        public CreateNotificationService(ICommandHandler commandHandler, IAsyncNotificationClient asyncNotificationClient, IQueryHandler queryHandler)
+        {
+            _commandHandler = commandHandler;
+            _asyncNotificationClient = asyncNotificationClient;
+            _queryHandler = queryHandler;
+        }
+
+        public async Task CreateEmailNotificationAsync(CreateEmailNotificationCommand notificationCommand, Dictionary<string, string> parameters)
+        {
+            var template = await _queryHandler.Handle<GetTemplateByNotificationTypeQuery, Template>(new GetTemplateByNotificationTypeQuery(notificationCommand.NotificationType));
+            if (template == null)
```

Comment:
I don't think the null check is necessary here. The validation will verify if the notification type is in the enum.
If the query can throw a `NotificationDalException` then the `ExceptionMiddleware` will kick in and return a bad request